protocol/consistencychecks.go

```diff
@@ -113,14 +123,42 @@ func (cc *ConsistencyChecks) updateSTR(requestType int, msg *Response) error {
 		// FIXME: check whether the STR was issued on time and whatnot.
 		// Maybe it has something to do w/ #81 and client transitioning between epochs.
 		// Try to verify w/ what's been saved
-		if err := cc.verifySTR(str); err == nil {
+		if err := verifySTR(cc.SavedSTR, str); err == nil {
 			return nil
 		}
 		// Otherwise, expect that we've entered a new epoch
 		if err := cc.verifySTRConsistency(cc.SavedSTR, str); err != nil {
 			return err
 		}
 
+	case MonitoringType:
+		// the requested epoch can be equal to saved epoch or the next expected epoch
+		// or epoch 0.
+		strs := msg.DirectoryResponse.(*DirectoryProofs).STR
+		switch {
+		case strs[0].Epoch == 0: /* prior history verification */
+			if err := verifySTR(cc.PinnedSTR, strs[0]); err != nil {
+				return err
+			}
+		case strs[0].Epoch == cc.SavedSTR.Epoch:
+			if err := verifySTR(cc.SavedSTR, strs[0]); err != nil {
+				return err
+			}
+		case strs[0].Epoch == cc.SavedSTR.Epoch+1:
+			if err := cc.verifySTRConsistency(cc.SavedSTR, strs[0]); err != nil {
+				return err
+			}
+		default:
+			panic("[coniks] Passt.")
```

Comment:
`Passt`? Sounds very [german](https://www.dict.cc/german-english/passt.html) ;-) I'm not sure what this is intended to say.